app/scripts/contentscript.js

```diff
@@ -261,10 +261,14 @@ const setupPageStreams = () => {
   pageChannel = pageMux.createStream(PROVIDER);
 };
 
+// The field below is used to ensure that replay is done only once for each restart.
+let REPLAY_ENABLED = false;
```

Comment:
It took me a little bit of time to wrap my head around this logic. Now I see that `REPLAY_ENABLED`, in other words, means something like canSendExtensionConnectedMessage

In the case that the listener receiving the message expands on the logic, this comment and variable name could become outdated. `REPLAY_ENABLED` works. I'm just mentioning this here in case we want to take a preventative and potentially more scalable approach by renaming this

Reply:
I'm back with a follow-up thought...

Essentially, this might just be me, but the naming here isn't sitting with me well. It seems to go from retry -> connect -> retry. 

In this PR, `REPLAY_ENABLED` is used to send the `METAMASK_EXTENSION_STREAM_CONNECT` message which is then mapped to a message for `retryOnMessage`

https://github.com/MetaMask/providers/blob/main/src/StreamProvider.ts
```
this._jsonRpcConnection = createStreamMiddleware({
  retryOnMessage: 'METAMASK_EXTENSION_STREAM_CONNECT',
});

```

what if we align the naming across the code to something like:
`CAN_SEND_RETRY_ACTIONS_ON_CONNECT_MSG`
`METAMASK_EXTENSION_RETRY_ACTIONS_ON_CONNECT`
`retryActionsOnConnectMessage`
?

Reply:
+1

I'd rename:
```
METAMASK_EXTENSION_STREAM_CONNECT => METAMASK_EXTENSION_CONNECT_CAN_RETRY
REPLAY_ENABLED => METAMASK_EXTENSION_CONNECT_SENT
retryOnMessage => retryActionsOnConnectMessage
```

Reply:
+1

I did renamed first 2 suggestions, changing the third is hard as it is used in 2 different repos.